tf/chunkparser.py

```diff
@@ -60,9 +64,10 @@ def chunk_reader(chunk_filenames, output_pipes):
             print("chunk_reader didn't find any chunks.")
             return None
         while len(chunks):
-            filename = chunks.pop()
-            done.append(filename)
-            output_pipes[n].send(filename)
+            if not pipe_full(output_pipes[n]):
+                filename = chunks.pop()
+                done.append(filename)
+                output_pipes[n].send(filename)
```

Comment:
If one of the children is failed - its probably good to know.  How often will it false positive if we have a log message in else path?

Reply:
I tested it and got multiple messages per second. Since reading filenames is so fast the pipes are almost always full.

Reply:
What do you think about registering an atexit to log that the subprocess is exiting?

Reply:
There should be exception printed to the console if a reader process crashes. It should also already print "Reader EOF" in `v4_gen` if a reader exits and the pipe is closed.

Reply:
That isn't what is happening at run2 training then.  Only standard log messages between training start and training stuck.

Reply:
Or child process I/O isn't being printed at all?